static/app/components/events/autofix/autofixOutputStream.tsx

```diff
@@ -0,0 +1,149 @@
+import {useEffect, useRef, useState} from 'react';
+import {keyframes} from '@emotion/react';
+import styled from '@emotion/styled';
+import {AnimatePresence, motion} from 'framer-motion';
+
+import type {AutofixStep} from 'sentry/components/events/autofix/types';
+import {IconArrow} from 'sentry/icons';
+import {space} from 'sentry/styles/space';
+import testableTransition from 'sentry/utils/testableTransition';
+
+interface Props {
+  step: AutofixStep;
+}
+
+const shimmer = keyframes`
+  0% {
+    background-position: -1000px 0;
+  }
+  100% {
+    background-position: 1000px 0;
+  }
+`;
+
+export function AutofixOutputStream({step}: Props) {
+  const [displayedText, setDisplayedText] = useState('');
+  const intervalRef = useRef<number>();
+  const previousText = useRef('');
+  const currentIndexRef = useRef(0);
+
+  useEffect(() => {
+    if (!step.output_stream) {
+      return undefined;
+    }
+
+    const newText = step.output_stream;
+
+    // Reset animation if the new text is completely different
+    if (!newText.startsWith(displayedText)) {
+      previousText.current = newText;
+      currentIndexRef.current = 0;
+      setDisplayedText('');
+    }
+
+    intervalRef.current = window.setInterval(() => {
+      if (currentIndexRef.current < newText.length) {
+        setDisplayedText(newText.slice(0, currentIndexRef.current + 1));
+        currentIndexRef.current++;
+      } else {
+        window.clearInterval(intervalRef.current);
+      }
+    }, 15);
+
+    return () => {
+      if (intervalRef.current !== undefined) {
+        window.clearInterval(intervalRef.current);
+      }
+    };
+  }, [displayedText, step.output_stream]);
+
+  if (!step.output_stream) {
```

Comment:
We're only using `output_stream` from the step right? Why not just pass only the output stream as a single prop, this way we can just not render this component at all if it doesn't exist